R/derive_param_lasta.R

```diff
@@ -0,0 +1,293 @@
+#'   for details)
+#'
+#'   *Default:* `NULL`
+#'
+#'   *Required or Optional:* Optional
+#'
+#' @param source_datasets Source dataframe to be used to calculate the
+#'                        first PD date
+#'
+#'   A named list of dataframes is expected (although for BoR) only one dataframe is
+#'   needed. It links the `dataset_name` from `source_pd` with an existing dataframe.
+#'
+#'   For example if `source_pd = pd_date` with
+#'   ```{r, eval=FALSE}
+#'   pd_date <- date_source(
+#'     dataset_name = "adrs",
+#'     date = ADT,
+#'     filter = PARAMCD == PD
+#'   )
+#'   ```
+#'   and the actual response dataframe in the script is `myadrs`, `source_datasets
+#'   = list(adrs = myadrs)` should be specified.
+#'
+#'    *Required or Optional:* Optional
+#'
+#' @param subject_keys Columns to uniquely identify a subject
+#'
+#'   A list of symbols created using `vars()`.
+#'
+#'   *Default:* `vars(STUDYID, USUBJID)`
+#'
+#'   *Required or Optional:* Required
+#'
+#' @param set_values_to Columns to set
+#'
+#'   A named list returned by `vars()` defining the columns to be set for the
+#'   new parameter, e.g. `vars(PARAMCD = "LSTAC", PARAM = "Last Disease
+#'    Assessment Censored at First PD by Investigator")` is expected. The values
+#'    must be symbols, character strings, numeric values, or `NA`.
+#'
+#'    *Required or Optional:* Required
+#'
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @examples
+#'
+#' library(dplyr)
+#' library(lubridate)
+#' library(admiral)
+#' library(tibble)
+#'
+#' adsl <- tribble(
+#'   ~USUBJID, ~TRTSDT,                      ~EOSDT,
+#'   "01",     ymd("2020-12-06"), ymd("2022-03-06"),
+#'   "02",     ymd("2021-01-16"), ymd("2022-02-03"),
+#'   "03",     ymd("2021-01-09"), ymd("2021-02-24"),
+#'   "04",     ymd("2021-04-21"), ymd("2021-09-15")
+#' ) %>%
+#'   mutate(STUDYID = "a_study_id")
+#'
+#' adrs <- tribble(
+#'   ~USUBJID, ~PARAMCD, ~AVAL, ~AVALC, ~ASEQ, ~ADT,
+#'   "01", "RSP", NA, "Y", 1, ymd("2021-04-08"),
+#'   "02", "RSP", NA, "N", 1, ymd("2021-05-07"),
+#'   "03", "RSP", NA, "N", 1, NA,
+#'   "04", "RSP", NA, "N", 1, NA,
+#'   "01", "PD", NA, "N", 1, NA,
+#'   "02", "PD", NA, "Y", 1, ymd("2021-05-07"),
+#'   "03", "PD", NA, "N", 1, NA,
+#'   "04", "PD", NA, "N", 1, NA,
+#'   "01", "OVR", 3, "SD", 1, ymd("2021-03-07"),
+#'   "01", "OVR", 2, "PR", 1, ymd("2021-04-08"),
+#'   "02", "OVR", 3, "SD", 1, ymd("2021-03-07"),
+#'   "02", "OVR", NA, NA, 1, ymd("2021-04-07"),
+#'   "02", "OVR", 6, "PD", 1, ymd("2021-05-07"),
+#'   "03", "OVR", 3, "SD", 1, ymd("2021-01-30"),
+#'   "03", "OVR", 3, "SD", 2, ymd("2021-01-30"),
+#'   "04", "OVR", NA, "NE", 1, ymd("2021-05-21"),
+#'   "04", "OVR", 5, "NON-PD", 1, ymd("2021-06-30"),
+#'   "04", "OVR", NA, "NE", 1, ymd("2021-07-24"),
+#'   "04", "OVR", NA, "ND", 1, ymd("2021-09-30"),
+#' ) %>%
+#'   mutate(STUDYID = "a_study_id")
+#'
+#' pd <- date_source(
+#'   dataset_name = adrs,
```

Comment:
This call fails because `dataset_name` expects a character value, i.e., `"adrs"` instead of `adrs`.